nexus-writer/src/hdf5_handlers/dataset.rs

```diff
@@ -1,3 +1,5 @@
+//! This module implements the [DatasetExt] and [HasAttributesExt] traits for
```

Comment:
A literal list of what is in the module is not overly helpful, if there is not a higher level description that is relevant then it is preferable to omit this.